clients/src/main/java/org/apache/kafka/clients/producer/ProducerRecord.java

```diff
@@ -56,6 +59,8 @@
     public ProducerRecord(String topic, Integer partition, Long timestamp, K key, V value) {
         if (topic == null)
             throw new IllegalArgumentException("Topic cannot be null");
+        if (!TopicValidator.isValidateTopicName(topic))
+            throw new InvalidTopicException(topic + " is not a valid topic name.");
```

Comment:
I originally wanted to use the logic from core's `Topic.scala`, however core isn't included as a dependency for clients.  Do you have any guidance on how to approach this?  I could see introducing a `common` module, and throwing in all the shared logic/code there, but that seems like a much broader conversation.